bin/lint-markdown-api-history.ts

```diff
@@ -0,0 +1,437 @@
+                  `'${breakingChangesFile}'\n\n` +
+                  'Parsed YAML:\n\n' +
+                  `${JSON.stringify(safeHistory, null, 4)}\n\n` +
+                  'API history block:\n\n' +
+                  `${possibleHistoryBlock.value}\n\n`,
+              );
+              errorCounter++;
+            }
+          }
+        }
+
+        // ? Maybe collect all api history and export it to a single file for future use.
+      }
+
+      // ? Maybe replace user YAML with result of <https://eemeli.org/yaml/#tostring-options> for consistent style (but not in CI)
+    }
+  } catch (error) {
+    errorCounter++;
+    console.error('Error occurred while linting:\n', error);
+  } finally {
+    return { historyBlockCounter, documentCounter, errorCounter, warningCounter };
+  }
+}
+
+function parseCommandLine() {
+  const showUsage = (arg?: string): boolean => {
+    if (!arg || arg.startsWith('-')) {
+      console.log(
+        'Usage: lint-roller-markdown-api-history [--root <dir>] <globs>' +
+          ' [-h|--help]' +
+          ' [--check-placement] [--breaking-changes-file <path>] [--check-strings]' +
+          ' [--schema <path>]' +
+          ' [--ignore <globs>] [--ignore-path <path>]',
+      );
+      process.exit(1);
+    }
+
+    return true;
+  };
+
+  const opts = minimist(process.argv.slice(2), {
+    boolean: ['help', 'check-placement', 'check-strings'],
+    string: ['root', 'ignore', 'ignore-path', 'schema', 'breaking-changes-file'],
+    unknown: showUsage,
+    default: {
+      'check-placement': true,
+      'check-strings': true,
+    },
+  });
+
+  if (opts.help || !opts._.length) showUsage();
+
+  return opts;
+}
+
+async function init() {
+  try {
+    const opts = parseCommandLine();
+
+    if (!opts.root) {
+      opts.root = '.';
+    }
+
+    if (opts.ignore) {
+      opts.ignore = Array.isArray(opts.ignore) ? opts.ignore : [opts.ignore];
+    } else {
+      opts.ignore = [];
+    }
+
+    if (opts['ignore-path']) {
+      const ignores = await readFile(resolve(opts['ignore-path']), { encoding: 'utf-8' });
+
+      for (const ignore of ignores.split('\n')) {
+        opts.ignore.push(ignore.trimEnd());
+      }
+    }
+
+    if (opts.schema) {
+      opts.schema = resolve(process.cwd(), opts.schema);
+      try {
+        await access(opts.schema, constants.F_OK | constants.R_OK);
+      } catch (error) {
+        throw new Error(
+          `Error occurred while attempting to access JSON Schema file: '${opts.schema}'\n${error}`,
+        );
+      }
+    }
+
+    if (opts['breaking-changes-file']) {
+      opts['breaking-changes-file'] = resolve(process.cwd(), opts['breaking-changes-file']);
+      try {
+        await access(opts['breaking-changes-file'], constants.F_OK | constants.R_OK);
+      } catch (error) {
+        throw new Error(
+          `Error occurred while attempting to access breaking changes file: '${opts['breaking-changes-file']}'\n${error}`,
+        );
+      }
+    }
```

Comment:
[Checking `fs.access` before reading is generally considered an anti-pattern in Node.js](https://nodejs.org/api/fs.html#fspromisesaccesspath-mode). In this case it would be better to let the error bubble up when a read actually happens, since the check doesn't prevent errors during the read itself.

Reply:
Resolved in [`ee64a4c` (#73)](https://github.com/electron/lint-roller/pull/73/commits/ee64a4ce04cdd7f7cefe81acf52263a5e88f42d0)